### ISSUE TITLE
tests only run the first time

### ISSUE THREAD

**Issue by dcsan** [rstacruz/tape-watch#155]
I'm using tape-watch. on saving my tests, the test gets fired and *something* runs again, but not the actual tests.

each subsequent run I just see:

```
1..0
# tests 0
# pass  0
```

any ideas where to look?

I am using 

`const test = require('tape-async');`

```
  tape-watch start +0ms [ undefined ]
  tape-watch requiring +137ms test/Wave-test.js
TAP version 13
# get files
  bot:Wave >> init start +0ms
  bot:Wave parsed: 17218 +5s
  bot:Wave wrote file +2ms
  bot:Wave Batch #0 has finished loading! +112ms
  bot:Wave << init done +56ms
  bot:Wave replyall: undefined +1ms
  bot:Wave input: filter on +0ms
  bot:Wave words: 2 chars: 9 +0ms
  bot:Wave length ok filter passed! filter on +0ms
  bot:Wave ribot:  +33ms
  bot:Wave replyall: undefined +0ms
  bot:Wave input: showfilter +0ms
  bot:Wave words: 1 chars: 10 +0ms
  bot:Wave length ok filter passed! showfilter +0ms
  bot:Wave ribot: inputfilter: on +6ms
  test:Wave r1: inputfilter: on +0ms
not ok 1 inputfilter: on
  ---
    operator: equal
    expected: false
    actual:   true
    at: Object.<anonymous> (/Users/dc/dev/rikai/ta-bot/kbot/test/Wave-test.js:20:7)
  ...
  tape-watch tape finished +5s

1..1
# tests 1
# pass  0
# fail  1
  tape-watch done +2ms
  tape-watch listening again +253ms
  tape-watch start +58s [ 'test/Wave-test.ts' ]
  tape-watch requiring +29ms test/Wave-test.js
TAP version 13
  tape-watch tape finished +5ms

1..0
# tests 0
# pass  0

# ok
  tape-watch done +0ms
  tape-watch listening again +253ms
  tape-watch start +119ms [ 'test/Wave-test.js' ]
  tape-watch requiring +7ms test/Wave-test.js
TAP version 13
  tape-watch tape finished +2ms

1..0
# tests 0
# pass  0

# ok
  tape-watch done +0ms
  tape-watch listening again +251ms
```

the tests make use of some async features with typescript

```
test('get files', async(t) => {
  let wave = new Wave()
  await wave.init()
  let r1 = wave.reply('TestBotGroup', 'filter on', 'username')
  let r2 = wave.reply('TestBotGroup', 'showfilter', 'username')
  debug("r1:",r2)
  t.equal(true, (r2 === "inputfilter: on"), r2 )

  let r3 = wave.reply('TestBotGroup', 'filter off', 'username')
  let r4 = wave.reply('TestBotGroup', 'showfilter', 'username')
  t.equal(true, (r4 === "inputfilter: on"), r2 )

  t.end()
});
```

**Comment by bholloway** [comment]
@dcsan I had the same issue when using `tape-bdd`. Hope this helps.

### TLDR

use `-R tape-async` and it might fix your problem.

Ensure the `-R` is **uppercase**.

### Long version

I hacked the `flush()` method at [`bin/tape-watch.js:136`](https://github.com/rstacruz/tape-watch/blob/master/bin/tape-watch#L136).

I added a 50% chance of deleting **any** `node_module`. I made sure to `console.log(fname)` of those that I did delete.

I then started triggering changes in my test files.

Now sometimes that crashed the test. Other times the tests did not rerun. However eventually the tests did rerun as desired.

At this point I copied the list of modules that I `console.log()`ed, before they were lost. I then changed the hack so that I was only deleting modules from that list, again with 50% chance. Something like:
```
const isDelete = [ ...files ].some(v => (v === fname)) && (Math.random() < 0.5);
```

Doing this about 5 times I zeroed in on the module that was causing the problem. For me that was `tape-bdd`.

Looking into the code in `flush()` I released that I could achieve the same thing without changing the source-code. Simply using `-R tape-bdd`.

**Comment by dcsan** [comment]
thanks for documenting!

whats the -R option for? 
cant get anything from the below

```
$ ./node_modules/tape/bin/tape -R
$ ./node_modules/tape/bin/tape -h
$ ./node_modules/tape/bin/tape --help
```

**Comment by rstacruz** [comment]
Try:

```
tape-watch --help
```

Which yields:

> -R, --refresh PACKAGE     ensure this PACKAGE gets refreshed

Thanks @bholloway for the well thought-out response!
